docusaurus/docs/recipes/contentful.md

```diff
@@ -0,0 +1,511 @@
+
+## Walk-through
+
+### 1. Installing Contentful SDK
+
+Using Contenful in your application can be done using [the **Contentful** SDK available as an npm package](https://www.npmjs.com/package/contentful), which you can install using the following command:
+
+```sh
+npm install --save contenful
+```
+
+This SDK provides access to **Contentful**'s [Delivery API](https://www.contentful.com/developers/docs/references/content-delivery-api/) and [Preview API](https://www.contentful.com/developers/docs/references/content-preview-api/), both of which will be used in the integration of **Contentful** in your app. 
+
+First, you'll need two **Contentful** access tokens and a space identifier. All three can be found in the Settings menu, under the `API keys` option:
+
+- Space ID,
+- Delivery token,
+- Preview token;
+
+Each token must be used together with their specific host, as follows:
+
+- Delivery host: `cdn.contentful.com`,
+- Preview host: `preview.contentful.com`;
+
+Setting up the SDK can be done as follows: 
+
+```js
+import { createClient } from 'contentful';
+
+const client = createClient({
+    space: // Space ID token
+    accessToken: // Either the Delivery token, or the Preview token
+    host: // Either the Delivery host, or the Preview host
+});
+```
+
+This `client` exposes the functions necessary to fetch the data in **Contentful**, for example:
+
+```js
+const result = await client.getEntries({ 
+    content_type: 'clients',
+});
+```
+
+If you're taking advantage of localization in your **Contentful** space, you must add to the request a field with your intended language code, like so:
+
+```js
+const result = await client.getEntries({
+    content_type: 'clients',
+    locale: 'en-US'
+});
+```
+
+### 2. Populating the app with fetched content
+
+To populate the app with the fetched content, we will explore two possible implementations: one for bigger applications that justify the inclusion of a Redux store, and another which doesn't.
+
+#### 2.1. Accessing the client directly though `getInitialProps`
+
+In Next.js apps, the `getInitialProps` function in general components typically receives a context object, which exposes many aspects of the context in which the page is rendered. Because we are building a custom app, and doing the work of running these `getInitialProps` functions ourselves, we can control what data they receive. In practice, we will be populating the context object with the information we want to propagate to our components. So, in your `App.js` file, you can check whether the page was loaded with a `cms-preview` query parameter, build your client, and share it across the app.
+
+This function also has the advantage letting you return props to your App function, making it easier to signal the App itself if you're in preview mode.
+
+```js
+// App.js
```

Comment:
This path should be the full like src/app/App.js. All other code blocks suffer from the same problem.